services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -46,7 +47,28 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
   override def getFileTree(project: Project, version: Option[Version])(
     implicit ec: ExecutionContext
   ): AsyncResult[List[String]] = ???
+
   override def getFile(project: Project, path: Path, version: Option[Version])(
     implicit ec: ExecutionContext
-  ): AsyncResult[String] = ???
+  ): AsyncResult[String] = {
+    val projectId: String = project.projectId.value
+    val filePath: String = URLEncoder.encode(path.toString, "UTF-8")
+    val fileVersion: String = version.map(el => el.value).getOrElse("master")
+
+    val responseFuture = httpClient.get(
+      s"${config.url}/projects/$projectId/repository/files/$filePath/raw",
```

Comment:
Here you have an error. You should use $project.repository here not $projectId.
And I think you should add a check for emptiness.

Reply:
It doesn't work this way

Reply:
Sorry, thanks!